src/main/kotlin/org/rust/ide/inspections/RsDeprecatedAttributeInspection.kt

```diff
@@ -0,0 +1,89 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.inspections
+
+import com.intellij.codeInspection.ProblemHighlightType
+import com.intellij.codeInspection.ProblemsHolder
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiElementVisitor
+import org.rust.lang.core.psi.RsFile
+import org.rust.lang.core.psi.RsMetaItem
+import org.rust.lang.core.psi.RsOuterAttr
+import org.rust.lang.core.psi.RsVisitor
+import org.rust.lang.core.psi.ext.*
+import org.rust.lang.core.psi.impl.RsModDeclItemImpl
+
+class RsDeprecatedAttributeInspection : RsLocalInspectionTool() {
+    override fun getDisplayName() = "Deprecated item"
+
+    override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean): PsiElementVisitor = object : RsVisitor() {
+        override fun visitElement(ref: RsElement) {
+            // non-inline module declaration
+            if (ref is RsModDeclItemImpl) return
+
+            ref.reference?.resolve()?.let {
+                val identifier = if (ref is RsReferenceElement) ref.referenceNameElement else ref
+                val original = if (it is RsFile) it.declaration ?: it else it
```

Comment:
Maybe it's better to get item where you want to search `deprecated` attribute first
and only after it call `checkAndRegisterAsDeprecated `
Note, you can't add `deprecated` attr in trait impl block
```rust
trait Foo {
    type Bar;
    const QWE: i32;
}

struct Baz;

impl Foo for Baz {
    type Bar = ();
    #[deprecated] // compilation error
    const QWE: i32 = 123;
}
```

Reply:
Agree, fixed